src/Perl6/Compiler.nqp

```diff
@@ -77,38 +68,16 @@ class Perl6::Compiler is HLL::Compiler {
             }
         }
 
-        my $*moreinput := sub ($cursor) {
-            my str $more := self.readline(nqp::getstdin(), nqp::getstdout(), '* ');
-            if nqp::isnull_s($more) || $more eq '' {
-                $more := ';';
-            }
-
-            $cursor."!APPEND_TO_ORIG"($more);
+        my $stdin    := nqp::getstdin();
+        my $encoding := ~%adverbs<encoding>;
+        if $encoding && $encoding ne 'fixed_8' {
+            nqp::setencoding($stdin, $encoding);
         }
 
-        my $super := nqp::findmethod(HLL::Compiler, 'interactive');
-        my $result := $super(self, :interactive(1), |%adverbs);
-        self.teardown();
+        my $result := $!p6repl.repl-loop(:interactive(1), |%adverbs);
```

Comment:
Keep in mind that the REPL6 load may have failed, and `$!p6repl` can be `Mu` here - we need to gracefully fallback if that's the case.